src/transformers/feature_extraction_utils.py

```diff
@@ -109,6 +115,102 @@ def values(self):
+            float_precision (`str`, *optional*):
+                The output floating point precision [float16, float32, double, bfloat16]
+        """
+        if (float_precision is None) or (tensor_type is None):
+            return self
+
+        # Convert to TensorType
+        if not isinstance(tensor_type, TensorType):
+            tensor_type = TensorType(tensor_type)
+
+        # Get a function reference for the correct framework
+        if tensor_type == TensorType.TENSORFLOW:
+            if not is_tf_available():
+                raise ImportError(
+                    "Unable to convert output to TensorFlow tensors format, TensorFlow is not installed."
+                )
+            import tensorflow as tf
+
+            target_framework = tf
+            cast_fun = tf.cast
+
+            def is_floating(x):
+                return x.dtype in (tf.float16, tf.float32, tf.double, tf.bfloat16)
+
+            is_tensor = tf.is_tensor
+
+        elif tensor_type == TensorType.PYTORCH:
+            if not is_torch_available():
+                raise ImportError("Unable to convert output to PyTorch tensors format, PyTorch is not installed.")
+            import torch
+
+            target_framework = torch
+
+            def cast_fun(x, dtype):
+                return x.to(dtype=dtype)
+
+            def is_floating(x):
+                return x.dtype in (torch.float16, torch.float32, torch.double, torch.bfloat16)
+
+            is_tensor = torch.is_tensor
+
+        # Jax tensors
+        elif tensor_type == TensorType.JAX:
+            if not is_flax_available():
+                raise ImportError("Unable to convert output to JAX tensors format, JAX is not installed.")
+            import jax.numpy as jnp  # noqa: F811
+
+            target_framework = jnp
+
+            def cast_fun(x, dtype):
+                return x.astype(dtype=dtype)
+
+            def is_floating(x):
+                return x.dtype in (jnp.float16, jnp.float32, jnp.double, jnp.bfloat16)
+
+            is_tensor = is_jax_tensor
+        # np arrays
+        else:
+            target_framework = np
+
+            def cast_fun(x, dtype):
+                return x.astype(dtype=dtype)
+
+            def is_floating(x):
+                return x.dtype in (np.half, np.single, np.double, np.longdouble)
+
+            is_tensor = is_numpy_array
+
+        if hasattr(target_framework, float_precision):
+            target_dtype = getattr(target_framework, float_precision)
+        else:
+            raise ValueError(
+                f"Failed to import the`dtype` {target_dtype} from the framework {target_framework} - please use a"
+                " supported",
+                " `dtype` for your targetted framework.",
+            )
+
+        # Do the tensor conversion in batch
+        for key, value in self.items():
+            # sanity check that we check for only tensors
+            if is_tensor(value):
+                if is_floating(value):
```

Comment:
I am not very comfortable to call these `is_tensor` and `if_floating` without checking if `value` is from `target_framework`.

For example, how about `tensor_type` being `pt` but `value` is a tf tensor?

Reply:
Thanks for the feedback! I guess this would not happen since the test is already done on the `convert_to_tensors` function that is called right before

Reply:
I agree with you @younesbelkada . However, this method is added like a public method, so the concern is there (despite I doubt any user will use it). If it is prefixed with `_`, I won't complain at all :-)

Let @sgugger review and give us his opinion if we should make any effort on such things.

Reply:
Ahh I see now! Yes then makes sense to have it prefixed with `_` 💪

Reply:
My opinion has been stated above. I don't think any of this is useful as Flax and TensorFlow deal differently with different dtypes, and there should only be a slight adaptation of the `to` method.